util/http/http_handler.cc

```diff
@@ -380,7 +382,7 @@ bool HttpConnection::CheckRequestAuthorization(const RequestType& req, HttpConte
       if (split_pos != std::string::npos) {
         std::string username = decoded.substr(0, split_pos);
         std::string password = decoded.substr(split_pos + 1, decoded.size());
-        if (username == "user" && password == owner_->password_) {
+        if (owner_->auth_functor_(path, username, password)) {
```

Comment:
this logic won't work to provide public access to  `/metrics`, for example.

Reply:
Why? I already tested this on DF and it works just fine? 

The callback would contain: 

```
[](auto path, auto username, auto password) {
  if(path == "/metrics") return true;
  //rest of auth here if the above branch is not true
}
```

https://github.com/dragonflydb/dragonfly/pull/1950/files#diff-1a804e4021a32c426f25ba8f4ad590c07f385dc90f01fe277fc9b8886ea90409R20

https://github.com/dragonflydb/dragonfly/pull/1950/files#diff-1a804e4021a32c426f25ba8f4ad590c07f385dc90f01fe277fc9b8886ea90409R20

Reply:
Is there something I am missing?

Reply:
you won't reach the callback because the header will be missing for regular requests. `authenticated` will stay false.

Reply:
you should check how come it passed your tests

Reply:
Nice catch! 
> you should check how come it passed your tests 
It passed the tests because I explicitly passed `username/password`. I added an extra test to cover this case. 

Again nice catch :pray:

Reply:
You could use the same API (i.e. with `path, user, pass`), and call it twice in the worst case. As long as the callback is fast, and it doesn't have any side effects, it might be an easier solution (or a simpler API)
But your call